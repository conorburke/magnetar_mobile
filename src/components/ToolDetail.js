import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';

class ToolDetail extends Component {
  render() {
    return (
      <View style={styles.container}>
        <View>
          <Text style={styles.header}>{this.props.toolDetails.title}</Text>
        </View>
        <View>
          <Text style={styles.detailFont}>{this.props.toolDetails.title}</Text>
          <Text style={styles.detailFont}>
            {this.props.toolDetails.category}
          </Text>
          <Text style={styles.detailFont}>
            {'$' + this.props.toolDetails.price}
          </Text>
          <Text style={styles.detailFont}>
            {this.props.toolDetails.first_name}
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  detailFont: {
    color: '#e4000f',
    fontSize: 15
  },
  header: {
    color: '#e4000f',
    fontSize: 30
  }
});

export default ToolDetail;
