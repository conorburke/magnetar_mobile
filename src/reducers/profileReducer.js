import { CREATE_PROFILE } from '../actions/types';

export default function(state = {}, action) {
    // console.log(action);
    switch (action.type) {
        case CREATE_PROFILE:
            return action.payload || state;
        default:
            return state
    }
}