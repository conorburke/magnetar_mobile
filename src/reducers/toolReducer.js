import { FETCH_TOOL } from '../actions/types';

export default function(state = [], action) {
    // console.log(action);
    switch (action.type) {
        case FETCH_TOOL:
            // console.log('action.payload', action.payload);
            // console.log('state', state);
            return action.payload || state;
            // return action.payload || false;
        default:
            return state
    }
}