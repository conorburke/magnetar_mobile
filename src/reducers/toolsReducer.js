import { FETCH_TOOLS } from '../actions/types';

export default function(state = [], action) {
    // console.log(action);
    switch (action.type) {
        case FETCH_TOOLS:
            // console.log('action.payload', action.payload);
            // console.log('state', state);
            return action.payload || state;
            // return action.payload || false;
        default:
            return state
    }
}