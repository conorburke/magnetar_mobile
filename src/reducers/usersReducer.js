import { FETCH_USERS } from '../actions/types';

export default function(state = [], action) {
    // console.log(action);
    switch (action.type) {
        case FETCH_USERS:
            // console.log('action.payload', action.payload);
            // console.log('state', state);
            return action.payload || state;
            // return action.payload || false;
        default:
            return state
    }
}