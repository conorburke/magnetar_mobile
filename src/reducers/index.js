import { combineReducers } from 'redux';

import authReducer from './authReducer'
import toolsReducer from './toolsReducer';
import usersReducer from './usersReducer';

export default combineReducers({
    auth: authReducer,
    tools: toolsReducer,
    users: usersReducer
});