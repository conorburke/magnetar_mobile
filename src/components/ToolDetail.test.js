import React from 'react';
import { shallow } from 'enzyme';
import ToolDetail from './ToolDetail';
import { createStore } from 'redux';
import reducers from '../reducers';

const store = createStore(reducers);

const toolDetails = {
  Title: "Ray's Hammer",
  Price: 1,
  ToolType: 'Hammer',
  FirstName: 'Randy'
};

const wrapper = shallow(
  <ToolDetail toolDetails={toolDetails} store={store} />
).dive();

it('should render the ToolDetail component', () => {
  expect(wrapper).toMatchSnapshot();
});
