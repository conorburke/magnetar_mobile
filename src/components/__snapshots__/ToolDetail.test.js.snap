// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ToolDetail component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "silver",
            "fontSize": 30,
          }
        }
      >
        Ray's Hammer
      </Text>
    </Component>
    <Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F5F5F5",
            "fontSize": 15,
          }
        }
      >
        Ray's Hammer
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F5F5F5",
            "fontSize": 15,
          }
        }
      >
        Hammer
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F5F5F5",
            "fontSize": 15,
          }
        }
      >
        $1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F5F5F5",
            "fontSize": 15,
          }
        }
      >
        Randy
      </Text>
    </Component>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Component>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "silver",
                "fontSize": 30,
              }
            }
          >
            Ray's Hammer
          </Text>
        </Component>,
        <Component>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            Ray's Hammer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            Hammer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            $1
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            Randy
          </Text>
        </Component>,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "silver",
                "fontSize": 30,
              }
            }
          >
            Ray's Hammer
          </Text>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Ray's Hammer",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "silver",
              "fontSize": 30,
            },
          },
          "ref": null,
          "rendered": "Ray's Hammer",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Ray's Hammer
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Hammer
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              $1
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Randy
            </Text>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Ray's Hammer",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": "Ray's Hammer",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Hammer",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": "Hammer",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "$1",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": "$1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Randy",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": "Randy",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "silver",
                  "fontSize": 30,
                }
              }
            >
              Ray's Hammer
            </Text>
          </Component>,
          <Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Ray's Hammer
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Hammer
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              $1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Randy
            </Text>
          </Component>,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-around",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "silver",
                  "fontSize": 30,
                }
              }
            >
              Ray's Hammer
            </Text>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Ray's Hammer",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "silver",
                "fontSize": 30,
              },
            },
            "ref": null,
            "rendered": "Ray's Hammer",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Ray's Hammer
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Hammer
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                $1
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Randy
              </Text>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Ray's Hammer",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "Ray's Hammer",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Hammer",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "Hammer",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "$1",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "$1",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Randy",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "Randy",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
